### DISCUSSION TITLE
MemberPress integration

### DISCUSSION THREAD

**Discussion by JollyCoderFr** [oidc-wp/openid-connect-generic#469] [Q&A]
Hello,

I've set up a Keycloak server and integrated it successfully with the plugin. However, MemberPress seems to have its own login page that ignores the new SSO rule implemented by your plugin. Any chance that you know how I can integrate this so that MemberPress login also goes through Keycloak ? Thanks in advance!

**Comment by timnolte** [comment]
So this would more than likely require an add-on for this plugin or a full code enhancement to provide that sort of support. This plugin is designed specifically to integrate with the standard WordPress login. Generally, the standard use case is focused on WordPress site management logins and not a lot of work has been done in the realm of membership style setups.

**Comment by timnolte** [comment]
This plugin does provide shortcodes for adding the login button or login URL elsewhere on a WordPress site so you may be able to just update the Member press login form template to use one of these shortcodes.

**Comment by JollyCoderFr** [comment]
Thanks for your answer. That's what I was thinking. I've been browsing the memberpress code trying to find how I can at least force it to use the default WP login but with no success. The only thing that is working for now is forcing the user to login if he tries to access certain memberpress pages, but I don't like it much because it's kinda messy and not easily maintainable if those pages become a lot.

Following your second suggestion, I wasn't able to find that form template to even edit it for now. All I found are the several classes used.

Any idea how I can use the login URL shortcode ?

**Reply by JollyCoderFr** [reply]
I dug deeper and I found that the login page url is most likely fetched from this function :

```
public function login_page_url($args = '') {
    if( isset($this->login_page_id) and
        is_numeric($this->login_page_id) and
        (int)$this->login_page_id > 0 ) {
      $link = MeprUtils::get_permalink($this->login_page_id);

      if(!empty($args)) {
        return $link.MeprAppCtrl::get_param_delimiter_char($link).$args;
      }
      else {
        return $link;
      }
    }

    return home_url(); // default to the home url
}
```

So I guess replacing the $link content with the SSO login url will work, don't you think?
